p2p/src/sync/index.rs

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+use crate::{
+    error::P2pError,
+    sync::{
+        mock_consensus::{BlockHeader, BlockId},
```

Comment:
There's already a BlockHeader defined in common

Reply:
I know. This is a bit of a can of worms because on one hand, having the mock block index implementation allows us to have self-contained P2P + syncing implementation that can be tested without connecting to other subsystem but if we want to make syncing generic over the chainstate it uses and chainstate generic over the P2P implementation it uses, it will be become complicated.